.github/workflows/changeset-review.yml

```diff
@@ -0,0 +1,87 @@
+name: Changeset Review
+
+on:
+  # Use pull_request_target to access secrets while keeping the workflow secure.
+  # We only checkout main (trusted code) and use GitHub API to inspect PR content.
+  pull_request_target:
+    types: [opened, synchronize, reopened]
+    paths:
+      - ".changeset/*.md"
+      - "packages/**"
+  # Manual trigger for testing
+  workflow_dispatch:
+    inputs:
+      pr_number:
+        description: "PR number to review"
+        required: true
+        type: number
+
+permissions:
+  contents: read
+  pull-requests: write
+  issues: write
+  id-token: write
+
+jobs:
+  review-changesets:
+    runs-on: ubuntu-latest
+    steps:
+      # Checkout main branch only - never checkout untrusted PR code
+      - name: Checkout base branch
+        uses: actions/checkout@v4
+
+      - name: Review Changesets with Claude
+        uses: anthropics/claude-code-action@beta
+        with:
+          anthropic_api_key: ${{ secrets.ANTHROPIC_API_KEY }}
+          direct_prompt: |
+            You are a changeset reviewer for the Cloudflare Workers SDK monorepo. Your job is to ensure changesets are correct, complete, and provide good changelog entries.
+
+            ## Guidelines
+
+            Read `.changeset/README.md` for the complete changeset guidelines. Use those rules to validate the changesets in this PR.
+
+            ## PR Information
+
+            PR Number: ${{ github.event.pull_request.number || inputs.pr_number }}
+
+            Use the GitHub CLI to inspect the PR. The `gh` command is available and authenticated.
+
+            Useful commands:
+            - `gh pr view <number> --json files,additions,deletions` - get list of changed files
+            - `gh pr diff <number>` - see the full diff
+            - `gh pr diff <number> -- '.changeset/*.md'` - see just changeset changes
+            - `gh api repos/${{ github.repository }}/contents/<path>?ref=<branch>` - fetch file from PR branch
+
+            ## Your Task
+
+            1. First, run `gh pr view` to get the list of changed files
+            2. Identify which packages under `packages/` were modified (excluding test files, fixtures)
+            3. Check what changeset files were added/modified using `gh pr diff`
+            4. Read the full content of any new changeset files from the PR branch
+
+            Then validate:
+
+            1. **Missing Changeset**: If packages were modified with user-facing changes, a changeset is required
```

Comment:
If this works well then could/should we remove the `no-changeset-required` label and associated CI checks?